File: backend/app/routers/notif.py
"""Notification-related API stubs (for the top notification bar)."""
from fastapi import APIRouter, Depends
from sqlalchemy import select, func
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.models.analysis import PracticalSelectRecord
from app.models.big_yang import BigYangAlert

router = APIRouter(prefix="/api", tags=["notifications"])


@router.get("/market-recaps/badge")
async def recap_badge():
    """Stub — Phase 2: implement daily market recap."""
    return {"latestId": None, "latestTradeDate": None, "today": 0, "yesterday": 0}


@router.get("/stock-analysis/list")
async def analysis_list(
    limit: int = 3,
    size: int = 3,
    db: AsyncSession = Depends(get_db),
):
    """Return recent practical-select records as analysis notifications."""
    stmt = (
        select(PracticalSelectRecord)
        .where(PracticalSelectRecord.status == "SUCCESS")
        .order_by(PracticalSelectRecord.submitted_at.desc())
        .limit(limit)
    )
    result = await db.execute(stmt)
    records = result.scalars().all()

    return {
        "records": [
            {
                "id": r.id,
                "stockName": r.stock_name,
                "stockCode": r.stock_code,
                "verdict": r.verdict,
                "summaryOneLiner": r.summary_one_liner,
                "status": r.status,
                "submittedAt": r.submitted_at.isoformat() if r.submitted_at else None,
            }
            for r in records
        ]
    }


@router.get("/tech-ai/alerts")
async def tech_ai_alerts():
    """Stub — Phase 2: implement AI monitoring alerts."""
    return []

File: frontend/src/stores/notif.js
import { defineStore } from 'pinia'
import { ref } from 'vue'

export const useNotifStore = defineStore('notif', () => {
  const items = ref([])
  const current = ref(0)

  async function fetchNotifications() {
    try {
      const [recap, analysis, bigyang, ai] = await Promise.allSettled([
        fetch('/api/market-recaps/badge').then(r => r.json()),
        fetch('/api/stock-analysis/list?limit=3').then(r => r.json()),
        fetch('/api/invest/big-yang/alerts').then(r => r.json()),
        fetch('/api/tech-ai/alerts').then(r => r.json()),
      ])

      const all = []

      if (recap.status === 'fulfilled' && recap.value?.latestId) {
        all.push({
          kind: 'recap',
          text: `${recap.value.latestTradeDate} 复盘已更新`,
          time: recap.value.latestTradeDate,
          href: '/market-recap',
        })
      }

      if (analysis.status === 'fulfilled' && analysis.value?.records) {
        for (const r of analysis.value.records) {
          all.push({
            kind: 'analysis',
            text: `${r.stockName} 已生成 · ${r.verdict || ''} · ${r.summaryOneLiner || ''}`,
            time: r.submittedAt,
            href: `/invest?analysis=${r.id}`,
          })
        }
      }

      if (bigyang.status === 'fulfilled' && Array.isArray(bigyang.value)) {
        for (const a of bigyang.value.slice(0, 3)) {
          all.push({
            kind: 'bigyang',
            text: `${a.stockCode} ${a.title || '大阳线告警'}`,
            time: a.triggerAt,
            href: '/invest',
          })
        }
      }

      items.value = all.sort((a, b) => new Date(b.time) - new Date(a.time)).slice(0, 8)
    } catch (e) {
      console.error('Failed to fetch notifications:', e)
    }
  }

  return { items, current, fetchNotifications }
})
